tests/FSharp.Test.Utilities/ProjectGeneration.fs

```diff
@@ -896,4 +1176,51 @@ type SyntheticProject with
     /// Saves project to disk and checks it with default options. Returns the FSharpChecker that was created
     member this.SaveAndCheck() =
         this.Workflow.Yield() |> Async.RunSynchronously |> ignore
-        this.Workflow.Checker
\ No newline at end of file
+        this.Workflow.Checker
+
+    static member CreateFromRealProject(projectDir) =
+
+        let projectFile =
```

Comment:
I was somewhat expecting to use to codepath which we already have for converting command line options into project options going into the checker.

i.e. by invoking a design time build, which would do all the msbuild magic, and end up with a list of fsc.exe CLI arguments.

As a benefit, that codepath would then be also tested by it.

Reply:
I think this is something @vzarytovskii wants to try to do. Should be doable but might have some other challenges.

Reply:
The downside is invoking msbuild to do an actual design time build, but I do not think that matters for benchmarking/testing, as this is a one time setup per project, can be cached.

I see a very big advantage in exercising code we already use in editor, and of course supporting a lot more different CLI options (translated from .fsbuild props) as well as configuration sources ;; most notable Directory.Build.props.